### PR TITLE
Refactor for inbound transfer

### PR DESCRIPTION
**Pull request by truthfool**
## Description

- Refactor for inbound transfer

 _(Ignore if these details are present on the associated JIRA ticket)_

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Design related bullet points or design document link related to this PR added in the description above. 

- [ ] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Add required Swagger annotation and update API documentation with details of any API changes if applicable

- [ ] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

### CODE REVIEW

**Comment by truthfool** [comment]
@avikganguly01 @fynmanoj

File: src/main/java/org/mifos/connector/ams/paygops/camel/route/PaygopsRouteBuilder.java [hunk 1]
```diff
@@ -210,22 +211,22 @@ public void configure() {
                 .log(LoggingLevel.INFO, "## Paygops user validation")
                 .setBody(e -> {
                     String body=e.getIn().getBody(String.class);
-                    logger.debug("Dfsp Id : {}",e.getProperty("dfspId"));
                     e.setProperty("dfspId",e.getProperty("dfspId"));
-                    logger.debug("Body : {}",body);
+                    logger.info("Body : {}",body);
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/ams/paygops/camel/route/PaygopsRouteBuilder.java]: use log level debug here

File: src/main/java/org/mifos/connector/ams/paygops/camel/route/PaygopsRouteBuilder.java [hunk 2]
```diff
@@ -144,12 +148,9 @@ public void configure() {
...
                         JSONObject paybillRequest = new JSONObject(exchange.getIn().getBody(String.class));
                         PaygopsRequestDTO paygopsRequestDTO = PayloadUtils.convertPaybillPayloadToAmsPaygopsPayload(paybillRequest);
-
-                        String transactionId = paygopsRequestDTO.getTransactionId();
                         log.info(paygopsRequestDTO.toString());
-                        exchange.setProperty(TRANSACTION_ID, transactionId);
+                        exchange.setProperty(TRANSACTION_ID, paygopsRequestDTO.getTransactionId());
                         exchange.setProperty("dfspId", exchange.getProperty("dfspId"));
-                        logger.info("DFSP Id:{}",exchange.getProperty("dfspId"));
                         logger.info("Validation request DTO: \n\n\n" + paygopsRequestDTO);
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/ams/paygops/camel/route/PaygopsRouteBuilder.java]: use debug level logs

File: src/main/java/org/mifos/connector/ams/paygops/camel/route/PaygopsRouteBuilder.java [hunk 3]
```diff
@@ -92,6 +92,10 @@ public void configure() {
                             exchange.setProperty("accountStatus",accountStatus.ACTIVE.toString());
                             exchange.setProperty("subStatus", "");
                             exchange.setProperty("dfspId", exchange.getProperty("dfspId"));
+                            exchange.setProperty(TRANSACTION_ID, exchange.getProperty(TRANSACTION_ID));
+                            exchange.setProperty("amount", result.getAmount());
+                            exchange.setProperty("currency", result.getCurrency());
+                            exchange.setProperty("msisdn", result.getWallet_msisdn().substring(1));
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/ams/paygops/camel/route/PaygopsRouteBuilder.java]: Why are we doing this substring without any condition check. look for the character you are trying to remove.

**Review by fynmanoj — CHANGES_REQUESTED** [review]

File: src/main/java/org/mifos/connector/ams/paygops/camel/route/PaygopsRouteBuilder.java [hunk 4]
```diff
@@ -91,7 +90,11 @@ public void configure() {
                             exchange.setProperty(PARTY_LOOKUP_FAILED, false);
                             exchange.setProperty("accountStatus",accountStatus.ACTIVE.toString());
                             exchange.setProperty("subStatus", "");
-                            exchange.setProperty("dfspId", exchange.getProperty("dfspId"));
+                            exchange.setProperty("accountHoldingInstitutionId", exchange.getProperty("accountHoldingInstitutionId"));
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/ams/paygops/camel/route/PaygopsRouteBuilder.java:93]: why do we need accountHoldingInstitutionId in paygops?

**Review by fynmanoj — CHANGES_REQUESTED** [review]
Clarification required

**Pull request merged**